ui/components/common/CodeEditor.js

```diff
@@ -22,9 +22,18 @@ const CodeEditor = class extends Component {
             // When the user presses tab we prevent the default propagation, so we need to udpate
             // the state manually here
             this.props.onChange(`${value.substring(0, selectionStart)}\t${value.substring(selectionEnd)}`);
+            this.updateCursorPosition(selectionStart + 1);
         }
     }
 
+    updateCursorPosition(position) {
+        const editor = this.editor.current;
+        this.setState({}, () => {
```

Comment:
What's setState here for?

Reply:
My thinking is: if the position of the cursor needs to be updated **after** the component's state is updated ('\t' character is added to textarea), one possible way to do this is via callback passed to `setState` function. This is kinda hacky since we're already updating the textarea with `setState` function triggered by `onChange` event [here](https://github.com/aerogear/data-sync-ui/blob/master/ui/components/schema/SchemaContainer.js#L39)

Honestly I would appreciate some suggestion from React domain experts on how to make this change more elegant and less hacky :P